### PR TITLE
Add test/UnionFind.hs

### PR DESCRIPTION
**Pull request by momohatt**

### CODE REVIEW

**Comment by momohatt** [comment]
@stevana Do you have any idea why the test is failing?

File: test/UnionFind.hs [hunk 1]
```diff
@@ -0,0 +1,230 @@
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  UnionFind
+-- Copyright   :  (C) 2017, ATS Advanced Telematic Systems GmbH
```
Comment (stevana) [inline test/UnionFind.hs]: Make it "(C) 2017-2019" and add your name here!

**Comment by momohatt** [comment]
Thank you!
I fixed the `Spec.hs` and the description, please check them.

File: test/Spec.hs [hunk 2]
```diff
@@ -80,6 +80,10 @@ tests docker0 = testGroup "Tests"
       , testProperty "parallel bad, see issue #218"
           (expectFailure (ioProperty (prop_echoParallelOK True <$> mkEnv)))
       ]
+  , testGroup "Union Find"
+      [ testProperty "sequential" prop_unionFind_sequential
```
Comment (stevana) [inline test/Spec.hs]: `import UnionFind` is missing?

Comment (stevana) [reply test/Spec.hs]: UnionFind should also be added to the .cabal file!

Comment (momohatt) [reply test/Spec.hs]: Right!! Thank you!

Comment (momohatt) [reply test/Spec.hs]: Sorry for multiple commits, I noticed your second message after I pushed my first commit

**Comment by stevana** [comment]
"This branch has conflicts that must be resolved" before we can merge!

File: test/Spec.hs (hunk 2 above)
Comment (stevana) [reply test/Spec.hs]: No problem :-)

**Comment by momohatt** [comment]
Somehow I can't see the details of the test results, what shall I do...?

**Comment by stevana** [comment]
That's strange. It seems to fail to compile with `stack test --pedantic`?

**Comment by momohatt** [comment]
```
[1 of 2] Compiling Main             ( /private/var/folders/hc/jrk9_nvx7lx3dw303jm5_7wr0000gn/T/stack14220/postgresql-libpq-0.9.4.2/Setup.hs, /private/var/folders/hc/jrk9_nvx7lx3dw303jm5_7wr0000gn/T/stack14220/postgresql-libpq-0.9.4.2/.stack-work/dist/x86_64-osx/Cabal-2.4.0.1/setup/Main.o )
[2 of 2] Compiling StackSetupShim   ( /Users/momohatt/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs, /private/var/folders/hc/jrk9_nvx7lx3dw303jm5_7wr0000gn/T/stack14220/postgresql-libpq-0.9.4.2/.stack-work/dist/x86_64-osx/Cabal-2.4.0.1/setup/StackSetupShim.o )
Linking /private/var/folders/hc/jrk9_nvx7lx3dw303jm5_7wr0000gn/T/stack14220/postgresql-libpq-0.9.4.2/.stack-work/dist/x86_64-osx/Cabal-2.4.0.1/setup/setup ...
Configuring postgresql-libpq-0.9.4.2...
setup: The program 'pg_config' is required but it could not be found.
```
No idea what to do...

**Comment by stevana** [comment]
Some tests use a postgresql database, you'll need `brew install libpq` or similar depending on your system (I'm guessing macos)?

**Comment by momohatt** [comment]
It worked (and yes I'm using MacOS), thank you so much!

**Comment by stevana** [comment]
Great, thanks for adding this nice example!

**Comment by momohatt** [comment]
Thanks for your review!

**Pull request merged**